railties/lib/rails/tasks/yarn.rake

```diff
@@ -10,13 +10,17 @@ namespace :yarn do
     end
 
     yarn_flags =
-      if `#{Rails.root}/bin/yarn --version`.start_with?("1")
+      if `"#{Rails.root}/bin/yarn" --version`.start_with?("1")
         "--no-progress --frozen-lockfile"
       else
         "--immutable"
       end
 
-    system({ "NODE_ENV" => node_env }, "#{Rails.root}/bin/yarn install #{yarn_flags}")
+    system({ "NODE_ENV" => node_env }, "\"#{Rails.root}/bin/yarn\" install #{yarn_flags}")
+  rescue Errno::ENOENT
+    $stderr.puts "bin/yarn was not found."
+    $stderr.puts "Please run `bundle exec rails app:update:bin` to create it."
+    exit 1
```

Comment:
Another issue #40795 popped up where apparently `bin/yarn` not exists. My assumption (to be confirmed) is that `rails app:update` was not run between Rails upgrades – which should create `bin/yarn`.

Therefore I added an error message to run `rails app:update:bin` in such cases where it should create `bin/yarn`.

---

One could also remove the `exit 1` in such cases so it only prints out the warning but the pipeline continues. It was like this in before Rails 6.1, but I'm not sure if this was the best idea because if Rails requires Yarn to be present it should hard fail if there is something wrong.